assets/src/stories-editor/components/post-selector/index.js

```diff
@@ -0,0 +1,339 @@
+			path: addQueryArgs( '/wp/v2/search', {
+				search,
+				per_page: 20,
+				subtype: searchablePostTypes.join( ',' ),
+			} ),
+		} ).then(
+			( suggestions ) => {
+				if ( this.isStillMounted && this.suggestionsRequest === suggestionsRequest ) {
+					this.setState( {
+						suggestions: map( suggestions, ( post ) => ( {
+							id: post.id,
+							title: decodeEntities( post.title ) || __( '(no title)', 'amp' ),
+							postType: post.subtype,
+						} ) ),
+						loading: false,
+					} );
+
+					if ( Boolean( suggestions.length ) ) {
+						this.props.debouncedSpeak( sprintf( _n(
+							'%d result found, use up and down arrow keys to navigate.',
+							'%d results found, use up and down arrow keys to navigate.',
+							suggestions.length, 'amp'
+						), suggestions.length ), 'assertive' );
+					} else {
+						this.props.debouncedSpeak( __( 'No results.', 'amp' ), 'assertive' );
+					}
+				}
+			}
+		).catch(
+			() => {
+				if ( this.isStillMounted && this.suggestionsRequest === suggestionsRequest ) {
+					this.setState( {
+						loading: false,
+					} );
+				}
+			}
+		);
+	}
+
+	updateSuggestions( value ) {
+		// Show the suggestions after typing at least 2 characters.
+		if ( value.length < 2 ) {
+			this.setState( {
+				showSuggestions: false,
+				selectedSuggestion: null,
+				loading: false,
+			} );
+
+			return;
+		}
+
+		this.setState( {
+			showSuggestions: true,
+			selectedSuggestion: null,
+			loading: true,
+		} );
+
+		this.fetchPostSuggestions( value );
+	}
+
+	onChange( event ) {
+		const inputValue = event.target.value;
+		this.props.onChange( inputValue );
+		this.updateSuggestions( inputValue );
+	}
+
+	onKeyDown( event ) {
+		const { showSuggestions, selectedSuggestion, suggestions, loading } = this.state;
+		// If the suggestions are not shown or loading, we shouldn't handle the arrow keys.
+		if ( ! showSuggestions || ! suggestions.length || loading ) {
+			switch ( event.keyCode ) {
+				// When UP is pressed, if the caret is at the start of the text, move it to the 0
+				// position.
+				case UP: {
+					if ( 0 !== event.target.selectionStart ) {
+						event.stopPropagation();
+						event.preventDefault();
+
+						// Set the input caret to position 0
+						event.target.setSelectionRange( 0, 0 );
+					}
+					break;
+				}
+				// When DOWN is pressed, if the caret is not at the end of the text, move it to the
+				// last position.
+				case DOWN: {
+					if ( this.props.value.length !== event.target.selectionStart ) {
+						event.stopPropagation();
+						event.preventDefault();
+
+						// Set the input caret to the last position
+						event.target.setSelectionRange( this.props.value.length, this.props.value.length );
+					}
+					break;
+				}
+				default:
+					break;
+			}
+
+			return;
+		}
+
+		const suggestion = this.state.suggestions[ this.state.selectedSuggestion ];
+
+		switch ( event.keyCode ) {
+			case UP: {
```

Comment:
For accessiblity, esc key key press should be handled here, exiting the dropdown.